### PR TITLE
Fixed #4906 : Peer review: "thank the contributor" should show snackb…

### LINKED ISSUE #4906: Peer review: "thank the contributor" should show snackbar instead of notification
adb

### PR DESCRIPTION
**Pull request by arinmodi**
Fixes #4906

What changes did you make and why?

Replaced notification with snackbar

**Tests performed**

Tested build variant on ASUS_X00TD with API level 28

Tested {build variant, e.g. ProdDebug} on {name of device or emulator} with API level {API level}.

### CODE REVIEW

**Comment by nicolas-raoul** [comment]
I just tried, this is pretty annoying, I feel like I can't press buttons because the snackbar just appeared...

@neslihanturan any UI advice on showing to the user that the thank was sent, without obstructing the UI? [screenshot](https://github.com/commons-app/apps-android-commons/issues/4906#issuecomment-1073226295)

**Comment by 4D17Y4** [comment]
> I just tried, this is pretty annoying, I feel like I can't press buttons because the snackbar just appeared...
> 
> @neslihanturan any UI advice on showing to the user that the thank was sent, without obstructing the UI? [screenshot](https://github.com/commons-app/apps-android-commons/issues/4906#issuecomment-1073226295)

I think `length short` would work just fine. It won't solve the issue of ui here but would not obstruct the ui for long. 
We can go with custom time if that feels too long.

**Comment by madhurgupta10** [comment]
IMO a toast would make better sense here thank snackbar since it doesn't no blocks the UI.

**Comment by arinmodi** [comment]
@nicolas-raoul, what i have to implement snackbar with length_short or toast

According to me, toast will be better

But @nicolas-raoul, please update
Thanks

**Comment by misaochan** [comment]
Another vote for toast here. :)

**Comment by nicolas-raoul** [comment]
If anyone agrees let's go with a short toast :-)

**Comment by arinmodi** [comment]
@nicolas-raoul, changes the snackbar to short lenght toast

File: app/src/main/java/fr/free/nrw/commons/review/ReviewController.java [hunk 1]
```diff
@@ -166,8 +168,7 @@ void sendThanks(@NonNull Activity activity) {
                         message = context.getString(R.string.send_thank_failure_message, media.getDisplayTitle());
                     }
 
-                    showNotification(title, message);
-
+                    ViewUtil.showShortToast(context,message);
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewController.java:171]: This statement was not covered in tests.
Add a test for this line ?

**Review by 4D17Y4 — CHANGES_REQUESTED** [review]
Looks good to me otherwise.

File: app/src/main/java/fr/free/nrw/commons/review/ReviewController.java (hunk 1 above)
Comment (arinmodi) [reply app/src/main/java/fr/free/nrw/commons/review/ReviewController.java:171]: @4D17Y4, how can i cover the code inside of the Observable.defer ?

Comment (4D17Y4) [reply app/src/main/java/fr/free/nrw/commons/review/ReviewController.java:171]: I believe it should be possible, can u look at the present instances in the codebase if there are any ? 
I think we have similar code but not sure if they are covered by tests. Let me know in any case.

Comment (arinmodi) [reply app/src/main/java/fr/free/nrw/commons/review/ReviewController.java:171]: @madhurgupta10, can you please help in this ?

Comment (madhurgupta10) [reply app/src/main/java/fr/free/nrw/commons/review/ReviewController.java:171]: I don't have access to my computer for a few days, maybe you can look at some samples in the codebase or maybe @devarsh-mavani-19 or @Ayan-10 can help you if they faced similar issues before.

Comment (devarsh-mavani-19) [reply app/src/main/java/fr/free/nrw/commons/review/ReviewController.java:171]: > I don't have access to my computer for a few days, maybe you can look at some samples in the codebase or maybe @devarsh-mavani-19 or @Ayan-10 can help you if they faced similar issues before.

I told him solution over call a few hours ago. As we were from same college previously and we are in touch with each other.

The way i handled it is by putting code of defer method in a separate function and writing tests for that function. i have implemented it previously in 1 of my PRs will update if i find it. Thanks

File: app/src/main/java/fr/free/nrw/commons/review/ReviewController.java [hunk 2]
```diff
@@ -156,19 +156,23 @@ void sendThanks(@NonNull Activity activity) {
...
-                        title = context.getString(R.string.send_thank_failure_title);
-                        message = context.getString(R.string.send_thank_failure_message, media.getDisplayTitle());
-                    }
+                    displayToast(context,result);
+                }, Timber::e);
+    }
 
-                    showNotification(title, message);
+    @SuppressLint("StringFormatInvalid")
+    private void displayToast(final Context context, final boolean result){
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewController.java:164]: displayToast is too generic. I would suggest `displayThanksToast` as this function is specific to the use case. 
Also add some documentation to this function.

**Review by 4D17Y4 — CHANGES_REQUESTED** [review]
Looks good otherwise, nice work.

**Comment by 4D17Y4** [comment]
I think this is related to :
https://github.com/commons-app/apps-android-commons/issues/1059

**Review by neslihanturan — APPROVED** [review]
Looks correct to me, thanks @arinmodi :) Just waiting tests to be done to merge

**Pull request merged**